src/server/session.ts

```diff
@@ -181,18 +181,29 @@ module ts.server {
         }
 
         semanticCheck(file: string, project: Project) {
-            var diags = project.compilerService.languageService.getSemanticDiagnostics(file);
-            if (diags) {
-                var bakedDiags = diags.map((diag) => formatDiag(file, project, diag));
-                this.event({ file: file, diagnostics: bakedDiags }, "semanticDiag");
+            try {
+                var diags = project.compilerService.languageService.getSemanticDiagnostics(file);

```

Comment:
I would prefer full name of the variable so that it is more consistent with other part of the codebase